src/promptflow-evals/promptflow/evals/evaluate/_evaluate.py

```diff
@@ -176,34 +224,32 @@ def evaluate(
     :rtype: ~azure.ai.generative.evaluate.EvaluationResult
     """
 
-    input_data_df = _validate_and_load_data(
-        target, data, evaluators, output_path, tracking_uri, evaluation_name)
+    input_data_df = _validate_and_load_data(target, data, evaluators, output_path, tracking_uri, evaluation_name)
+
+    # Process evaluator config to replace ${target.} with ${data.}
+    evaluator_config = _process_evaluator_config(evaluator_config)
+    _validate_columns(input_data_df, evaluators, target, evaluator_config)

```

Comment:
Is this check for fail early ? I see same happening later when target gets calculated.

Reply:
Yes, this check will fail early if requires columns are missing.

Reply:
Two cases:

1. If target is None, this check will fail immediately if missing column found by comparing to evaluator signature.
2. If target is not None, the check will be performed in two steps, 
 - Here, it ensure data has the required column for target to run
 - Later on, it performs a full check against combined dataset (data + target)